payments/schema/mutations.py

```diff
@@ -614,6 +620,70 @@ def mutate_and_get_payload(cls, root, info, **input):
         return ApproveOrderMutation(failed_orders=failed_orders)
 
 
+class ResendOrderMutation(graphene.ClientIDMutation):
+    class Input:
+        orders = graphene.List(graphene.NonNull(graphene.ID))
+        due_date = graphene.Date(description="Defaults to the Order due date")
+        profile_token = graphene.String(
+            required=False, description="API token for Helsinki profile GraphQL API",
+        )
+
+    sent_orders = graphene.List(graphene.NonNull(graphene.ID), required=True)
+    failed_orders = graphene.List(graphene.NonNull(FailedOrderType), required=True)
+
+    @classmethod
+    @change_permission_required(Order)
+    def mutate_and_get_payload(cls, root, info, **input):
+
+        orders = []
+        due_date = input.pop("due_date", None)
+        for order_id in input["orders"]:
+            order = get_node_from_global_id(
+                info, order_id, only_type=OrderNode, nullable=False
+            )
+            orders.append(order)
+
+            if order.lease.status != LeaseStatus.OFFERED:
+                VenepaikkaGraphQLError(
+                    _(
+                        "Cannot resend an invoice for a lease that is not currently offered."
+                    )
+                )
+
+        failed_orders = []
+        sent_orders = []
+
+        # update order details in a loop first, before sending a single invoice, so
+        # it's less likely that the invoicing operation fails halfway through
+        for order in orders:
+            if not order.customer_email and not order.customer_phone:
+                if not input.get("profile_token"):
+                    raise VenepaikkaGraphQLError(
+                        _(
+                            "Profile token is required if an order does not previously have email or phone."
+                        )
+                    )
+                profile = fetch_order_profile(order, input["profile_token"])
+                update_order_from_profile(order, profile)
```

Comment:
Now that I'm thinking about it, if _one_ order does not have the customer and all the rest do, it will make the whole thing fail. So it would probably be better to add it to the failed orders and add the token error as message 🤔 
```suggestion
        token = input.get("profile_token")
        for order in orders:
            if not order.customer_email and not order.customer_phone:
                if token:
                    profile = fetch_order_profile(order, input["profile_token"])
                    update_order_from_profile(order, profile)
                else:
                    failed_orders.append(
                        FailedOrderType(
                            id=order_id,
                            error=_("Profile token is required if an order does not previously have email or phone.")
                        )
                    )
                    continue
                
```

Reply:
hmm yes...

Reply:
on the other hand, it's nice if a operation either fails or succeed, not half and half... but it will be frustrating for admins if one missing contact info causes multiple invoices to fail.